notification/expression.go

```diff
@@ -30,38 +30,49 @@ type ExpressionRunner struct {
 	CelEnv       map[string]any
 }
 
-func (t ExpressionRunner) logToJobHistory(ctx api.Context, name, errMsg string) {
-	jobHistory := models.NewJobHistory(name, t.ResourceType, t.ResourceID)
-	jobHistory.Start()
-	jobHistory.AddError(errMsg)
-	if err := db.PersistJobHistory(ctx, jobHistory.End()); err != nil {
-		logger.Errorf("error persisting job history: %v", err)
-	}
-}
-
 // Eval evaluates the given expression into a boolean.
 // The expression should return a boolean value that's supported by strconv.ParseBool.
 func (t ExpressionRunner) Eval(ctx api.Context, expression string) (bool, error) {
+	jobHistory := models.NewJobHistory("NotificationFilterEval", t.ResourceType, t.ResourceID).Start()
```

Comment:
@moshloop made the change. One issue we still have is how a notification can fail for some events but pass for some other event.

Eg: A notification with filter `check.type = 'http'` might listen on events
- `incident.created` 
- `check.passed`

This could be considered an incorrectly setup notification but that's still allowed. If you have no filter in this case, then that's a perfectly working notification.
Maybe in the UI we should not allow mixing events that have different template variables.